### PR TITLE
Add github action for build_docker on merge

### LINKED ISSUE #86: Push docker images through post-merge CI pipeline
Currently docker images are built and pushed through `./build_docker.sh` but this requires somebody to run this outside of CI. We should hook this up with credentials in github actions to ensure that images for master are always built and pushed.

### PR DESCRIPTION
**Pull request by Qinusty**
This allows for docker images to be built and pushed via github actions. Resolves #86

@oliverchang We will need to add the necessary secrets via the repository configuration for pushing to gcr.io

### CODE REVIEW

**Comment by Qinusty** [comment]
analysis dockerfile `podman pull` has been changed to use `vfs` as the storage driver, without this the action encountered an error relating to the mounting of the aufs storage folder as seen [here](https://github.com/ossf/package-analysis/runs/2774693559?check_suite_focus=true).

This continues to work fine for me locally.

File: analysis/Dockerfile [hunk 1]
```diff
@@ -29,6 +29,6 @@ COPY --from=build /src/analyze /usr/local/bin/analyze
 COPY --from=build /src/worker /usr/local/bin/worker
 
 # Cache analysis images.
-RUN podman pull gcr.io/ossf-malware-analysis/node && \
-    podman pull gcr.io/ossf-malware-analysis/python && \
-    podman pull gcr.io/ossf-malware-analysis/ruby
+RUN podman --storage-driver vfs pull gcr.io/ossf-malware-analysis/node && \
```
Comment (oliverchang) [inline analysis/Dockerfile:32]: Per https://docs.docker.com/storage/storagedriver/select-storage-driver/ vfs does not seem like a recommended thing to use.

(I'm assuming this applies to podman too)

**Review by oliverchang — COMMENTED** [review]
Thanks! given the issue with the vfs though, I can try to set this up as a gcp cloudbuild instead and if that goes away. That way we also avoid having to manage secrets.

File: analysis/Dockerfile (hunk 1 above)
Comment (Qinusty) [reply analysis/Dockerfile:32]: vfs was admittedly the first driver I managed to get working both locally and remotely. Cloudbuild seems like the optimal solution here and will hopefully remove any issues surrounding the storage driver during the build of the analysis image.

**Comment by oliverchang** [comment]
I tried this on cloudbuild, but get a similar aufs error. Will look into a bit more next week.

**Comment by Qinusty** [comment]
> I tried this on cloudbuild (#89), but get a similar aufs error. Will look into a bit more next week.

I'd hazard a guess that this is related to podman being ran within several containerised layers (CI container, docker build container) with some permissions issues from the nonroot distroless container. I'll leave this PR open as a discussion point until we resolve this.

Additionally, even without the push credentials we probably want the containers to be built as part of the PR CI process. If we fix the aufs issues for cloudbuild we should aim to translate those fixes into [this](https://github.com/ossf/package-analysis/pull/88/commits/2462770106c13eafdd6b3dc88694b756b19b8782#diff-a6ce9b5b0a53d8dc6dbd59d40b23994c3e68a91fac73c0d7eacdf373c95476c9) at least.

**Comment by oliverchang** [comment]
Hey @Qinusty sorry for the delay here.

I'm still not sure how to fix the permission issue (we probably need something similar to `--privileged`, but this isn't available for `docker build`).

In the meantime, to unblock this PR, we can do something like:

```
ARG NO_PODMAN_PULL
RUN test -n "$NO_PODMAN_PULL" || \
    (podman pull gcr.io/ossf-malware-analysis/node && \
     podman pull gcr.io/ossf-malware-analysis/python && \
     podman pull gcr.io/ossf-malware-analysis/ruby)
```

Then build with `docker build --build-arg NO_PODMAN_PULL=1 ...` on CI. 

We can also remove `--squash` from the docker build arguments, since we don't need it anymore. We squash[ inside the Dockerfile](https://github.com/ossf/package-analysis/blob/560f3b08f3836e9778e41f742eeba96227c38272/build/python/Dockerfile#L7) in the relevant images instead.

**Comment by Qinusty** [comment]
That makes sense for the build phase as we're not looking to push these images here, they don't need to be pre-loaded with the individual analysis images. I'll get this pushed up later today.

File: .github/workflows/build-images.yml [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v2
+
+    - name: setup-go
+      uses: actions/setup-go@v2
+      with:
+        go-version: '1.15.5'
+
+    - name: Enable docker experimental
```
Comment (oliverchang) [inline .github/workflows/build-images.yml:17]: is this still needed?

**Review by oliverchang — APPROVED** [review]
thanks a lot! Just one minor comment that can come in another PR if necessary.

**Pull request merged**